### ISSUE TITLE
[carddav] Problem with 'carddav_protect'

### ISSUE THREAD

**Issue by GoogleCodeExporter** [roundcubevnz/myroundcube#457]
```
I running:

- Roundcube 0.8.4
- Plugin Manger 8.8.4
- carddav 3.2.1
- carddav_plus 1.0.1

My users should use only carddav addressbooks, the default sql addressbook 
shouldn't be displayed, though.

If I set "$rcmail_config['carddav_protect'] = false;" the default sql 
addressbook is shown.

If I set "$rcmail_config['carddav_protect'] = true;" the default sql 
addressbook and the option "CardDav Settings" under 
"Settings->Preferences->Manage Plugins" is hidden. BTW, Shouldn't be this 
option displayed under "Settings->Preferences->Addressbook" ?

For me the shown sql addressbook is a bug and should be fixed. Agree?

The other thing is that a user shouldn't be able to edit the default carddav 
addressbooks but should add other addressbooks as he want.

Is this also a bug or the default behaviour?

Thanks in advance!


```

Original issue reported on code.google.com by `tob...@hachmer.de` on 2 Jan 2013 at 11:49

**Comment by GoogleCodeExporter** [comment]
```
There is an option now to hide (empty) default addressbooks (see changelog).

I leave the ticket open until next major release is out.
```

Original comment by `roland.l...@gmail.com` on 3 Jan 2013 at 9:44
- Changed state: **Accepted**

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the quick assistance!
But for my understanding the carddav plugin behaves strange. What I want for my 
users is:

- two default addressbooks which they can't edit
- opportunity to add additional individual addressbooks
- the default sql addressbook needn't be displayed anywhere, not under 
addressbooks and not under the compose addressbook pane

I downloaded and tested the carddav plugin version 3.2.1.
I added my config below.
The behaviour:

- CardDAV Settings is not shown below Addressbook (see attached screenshot; 
carddav_section.jpg)

"$rcmail_config['carddav_protect'] = false;" && complete new user (deleted user 
from roundcube sql table "users")
- the empty sql addressbook is shown under addressbooks (see attached 
screenshot; carddav_notprotect_defaultabook.jpg)
- The setting "Add new contacts to the selected addressbook" is set by default 
to the empty sql addressbook. If I choose here one of the carddav addressbooks 
and save the setting the default sql addressbook disappears under addressbooks. 
Just like I expected. 
--> Is this a bug or should there be a setting to set the default for "Add new 
contacts to the selected addressbook", also?

"$rcmail_config['carddav_protect'] = true;" && complete new user (deleted user 
from roundcube sql table "users")
- the empty sql addressbook is shown under addressbooks AND one of my two 
default carddav addressbooks is missing (see attached screenshot; 
carddav_protect_defaultabook_onecarddavmissing.jpg)
- the setting "Add new contacts to the selected addressbook" is set by default 
to the empty sql addressbook. If I choose here the carddav addressbook (the 
only on, the carddav plugin has two configured) and save the setting the 
default sql addressbook disappears under addressbooks. No effect to the carddav 
abooks. Only one is shown and not the two configured.

- When composing a new email the default sql addressbook is shown in the left 
pane, either. Is there way to hide this here also or another bug?

Thanks in advance!


<?php
/* automatic addressbook database table name */
$rcmail_config['db_table_collected_contacts'] = 'collected_contacts';

/* Is automatic_addressbook enabled or disabled by default for users? */
$rcmail_config['use_auto_abook'] = false;

/* Should automatic_addressbook be used for completion by default? */
$rcmail_config['use_auto_abook_for_completion'] = false;

/* Show database addressbook even if they are empty */
$rcmail_config['show_empty_database_addressbooks'] = false;

/* CardDAV database tables */
$rcmail_config['db_table_carddav_server'] = 'carddav_server';
$rcmail_config['db_table_carddav_contacts'] = 'carddav_contacts';

/* Debug (extended logging) */
$rcmail_config['carddav_debug'] = true;

/* Prevent users from editing defaults */
$rcmail_config['carddav_protect'] = false;

/* Max CardDAVs */
$rcmail_config['max_carddavs'] = 5;

/* Sync CardDAVs every (x) minutes automatically (0 = on login only) */
$rcmail_config['sync_carddavs_interval'] = 5;

/* Slow servers
   CAUTION: This method is not failsafe but speeds up things a little bit */
$rcmail_config['carddav_slow_backends'] = array(
  'google.com' => 5, // assume vcard created/updated after (x) secs.
);

/* Host where passwords are in sync with IMAP passwords */
$rcmail_config['carddav_synced_passwords'] = array(
);

/* Default CardDAV addressbooks
   placeholders: %u = $_SESSiON['username'], %p = $_SESSION['password']
*/
$rcmail_config['def_carddavs'] = array(
  'Persönliches Adressbuch' => array(
                   'user' => '%u',
                   'pass' => '%p',
                   'url' => 'http://example.com/caldav.php/%u/addresses/',
                   'readonly' => false,
                 ),
  '2nd Adressbuch' => array(
                   'user' => '%u',
                   'pass' => '%p',
                   'url' => 'http://example.com/caldav.php/2nd@example.com/addresses/',
                   'readonly' => false,
                 ),
);
?>
```

Original comment by `tob...@hachmer.de` on 3 Jan 2013 at 12:49

Attachments:
- [carddav_notprotect_defaultabook.jpg](https://storage.googleapis.com/google-code-attachments/myroundcube/issue-457/comment-2/carddav_notprotect_defaultabook.jpg)
- [carddav_section.jpg](https://storage.googleapis.com/google-code-attachments/myroundcube/issue-457/comment-2/carddav_section.jpg)
- [carddav_protect_defaultabook_onecarddavmissing.jpg](https://storage.googleapis.com/google-code-attachments/myroundcube/issue-457/comment-2/carddav_protect_defaultabook_onecarddavmissing.jpg)

**Comment by GoogleCodeExporter** [comment]
```
Just to make sure, we are on the same page ... Do you have installed only 
carddav and carddav_plus (required for advanced features)?
```

Original comment by `roland.l...@gmail.com` on 3 Jan 2013 at 1:00

**Comment by GoogleCodeExporter** [comment]
Original comment by `roland.l...@gmail.com` on 3 Jan 2013 at 1:00
- Added labels: **Priority-Low**
- Removed labels: **Priority-High**

**Comment by GoogleCodeExporter** [comment]
```
What do you mean with only?
Here the list of my plugin_manager config, For my tests I activated all plugins 
in global section:

  'globalplugins' => array(
    'jqueryui' => array(
      'protected' => true,
      'active' => true
    ),
    'qtip' => array(
      'protected' => true,
      'active' => true
    ),
    'settings' => array(
      'protected' => true,
      'active' => true
    ),
    'password' => array(
      'protected' => true,
      'active' => true
    ),
    'timepicker' => array(
      'protected' => true,
      'active' => true
    ),
    'carddav' => array(
      'protected' => true,
      'active' => true
    ),
    'calendar' => array(
      'protected' => true,
      'active' => true
    ),
    'carddav_plus' => array(
      'protected' => true,
      'active' => true
    ),
    'calendar_plus' => array(
      'protected' => true,
      'active' => true
    ),
    'acl' => array(
      'protected' => true,
      'active' => true
    ),
    'show_additional_headers' => array(
      'protected' => true,
      'active' => true
    ),
    'compose_in_taskbar' => array(
      'protected' => true,
      'active' => true
    ),
    'compose_newwindow' => array(
      'protected' => true,
      'active' => true
    ),
    'contextmenu' => array(
      'protected' => true,
      'active' => true
    ),
    'copymessage' => array(
      'protected' => true,
      'active' => true
    ),
    'emoticons' => array(
      'protected' => true,
      'active' => true
    ),
    'jscolor' => array(
      'protected' => true,
      'active' => true
    ),
    'listcommands' => array(
      'protected' => true,
      'active' => true
    ),
    'global_alias' => array(
      'protected' => true,
      'active' => true
    ),
    'http_auth' => array(
      'protected' => true,
      'active' => true
    ),
    'http_request' => array(
      'protected' => true,
      'active' => true
    ),
    'idle_timeout' => array(
      'protected' => true,
      'active' => true
    ),
    'userinfo' => array(
      'protected' => true,
      'active' => true
    ),
    'zipdownload' => array(
      'protected' => true,
      'active' => true
    ),
  ),

```

Original comment by `tob...@hachmer.de` on 3 Jan 2013 at 1:09

**Comment by GoogleCodeExporter** [comment]
```
caldav_protect has a bug. Confirmed.

Sorting order works for me. You don't have to register helper plugins. They are 
loaded automatically if and when required (jqueryui, qtip, settings, 
timepicker, jscolor, global_alias, http_auth).
```

Original comment by `roland.l...@gmail.com` on 3 Jan 2013 at 2:44
- Added labels: **Priority-Medium**
- Removed labels: **Priority-Low**

**Comment by GoogleCodeExporter** [comment]
```
Please update and post feedback. Thanks!
```

Original comment by `roland.l...@gmail.com` on 3 Jan 2013 at 3:06

**Comment by GoogleCodeExporter** [comment]
```
Thanks for the bugfix. Now with $rcmail_config['carddav_protect'] = true; the 
two carddav addressbooks are listed. And thanks for the hints with the helper 
plugins!

Do I have to create a feature request for my other issues: ?

1. I want to set two default (or a defined set of) carddav addressbooks which 
the user can't edit, but the opportunity for him to add other carddavs.
2. The setting "Add new contacts to the selected addressbook" is set by default 
to the empty sql addressbook. If I choose here the carddav addressbook and save 
the setting the default sql addressbook disappears under addressbooks. There 
should also be a configurable default setting for that.

Regards and thanks in advance!
```

Original comment by `tob...@hachmer.de` on 3 Jan 2013 at 5:50

**Comment by GoogleCodeExporter** [comment]
```
And I forget to mention that the default sql addressbook is listed in contacts 
pane when composing a new email. Is this a roundcube thing or belongs to this 
plugin?
```

Original comment by `tob...@hachmer.de` on 3 Jan 2013 at 5:53

**Comment by GoogleCodeExporter** [comment]
```
This ticket will be open until compose screen glitch is fixed.
I have to investigate Roundcube Core why it does not run into my routine to 
unset empty addressbooks.

For everything else please open separate tickets. Please use Plugin Manager 
Center "Issue" link nearby the plugin. It makes my life easier. I won't have to 
ask about your plugin settings. I can detect your settings from database. 
Nothing to worry about, I only see the plugins you use, your database backend, 
your PHP version, your Roundcube version and the way how you installed your 
plugins.

Thanks!
```

Original comment by `roland.l...@gmail.com` on 3 Jan 2013 at 6:12

**Comment by GoogleCodeExporter** [comment]
```
Fixed in 3.2.3.
```

Original comment by `roland.l...@gmail.com` on 4 Jan 2013 at 5:11
- Changed state: **Fixed**

**Comment by GoogleCodeExporter** [comment]
```
Thanks very much, it has worked!!!
```

Original comment by `tob...@hachmer.de` on 4 Jan 2013 at 10:59
